File: educational-system-consumer-admin-9000/src/main/java/edu/hubu/consumeradmin/service/IRoleService.java
package edu.hubu.consumeradmin.service;

import edu.hubu.commons.model.CommonResult;
import edu.hubu.commons.model.admin.Role;
import org.springframework.cloud.openfeign.FeignClient;
import org.springframework.web.bind.annotation.*;

/**
 * @author moonlan
 * date 2021/2/1 下午4:54
 */
@SuppressWarnings(value = "unused")
@FeignClient(name = "admin-provider-service")
public interface IRoleService {

    @GetMapping("/role/{id}")
    CommonResult selectOne(@PathVariable("id") String id);

    @GetMapping("/role/all")
    CommonResult selectAll(@RequestParam(value = "orderBy", defaultValue = "roleId") String orderBy);

    @GetMapping("/role/")
    CommonResult selectAllPage(
            @RequestParam(value = "currentPage", defaultValue = "1") Integer currentPage,
            @RequestParam(value = "pageSize", defaultValue = "5") Integer pageSize,
            @RequestParam(value = "orderBy", defaultValue = "roleId") String orderBy
    );

    @PostMapping("/role/select-like")
    CommonResult selectLikePage(
            @RequestParam(value = "currentPage", defaultValue = "1") Integer currentPage,
            @RequestParam(value = "pageSize", defaultValue = "5") Integer pageSize,
            @RequestParam(value = "orderBy", defaultValue = "roleId") String orderBy,
            @RequestBody edu.hubu.commons.mybatis.model.Role role
    );

    @PostMapping("/role/")
    CommonResult save(@RequestBody Role role);

    @PutMapping("/role/{id}")
    CommonResult update(@PathVariable("id") String id, @RequestBody Role role);

    @DeleteMapping("/role/{id}")
    CommonResult logicDeleteById(@PathVariable("id") String id);

    @PatchMapping("/role/{id}")
    CommonResult forbidById(@PathVariable("id") String id);

    @DeleteMapping("/role/delete/{id}")
    CommonResult deleteById(@PathVariable("id") String id);
}

File: educational-system-commons/src/main/java/edu/hubu/commons/service/admin/IRoleService.kt
package edu.hubu.commons.service.admin

import edu.hubu.commons.model.admin.Role;

/**
* @author moonlan
* date 2021/1/25 下午2:07
*/
interface IRoleService {
    fun save(t: Role): Role
    fun logicalDeleteById(id: String): Boolean
    fun deleteById(id: String): Boolean
    fun updateById(id: String, t: Role): Role
    fun forbidById(id: String): Boolean
    fun selectOneById(id: String): Role
    fun selectAllOrderBy(order: String = "roleId"): List<Role>
    fun selectAllPageOrderBy(currentPage: Int = 1, pageSize: Int = 5, order: String = "roleId"): List<Role>
    fun selectLikePageOrderBy(currentPage: Int = 1, pageSize: Int = 5, order: String = "roleId", t: edu.hubu.commons.mybatis.model.Role): List<edu.hubu.commons.mybatis.model.Role>
}
